egl: Compile failure as eglextchromium.h replaced by eglext_angle.h in mesa git

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
_Here is a clear and concise description of what the problem is:_
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
With recent mesa git, its commit [egl: Remove eglextchromium.h and import eglext_angle.h](https://gitlab.freedesktop.org/mesa/mesa/-/commit/f5bb9dd738ace274c97507adea073b6c609469b2) has done what it states and removed the eglextchromium.h header causing  Kodi to fail to compile xbmc/windowing/X11/GLContextEGL.cpp which used that header.

## Expected Behavior
_Here is a clear and concise description of what was expected to happen:_
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
Compiles.


## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
Does not compile.


## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->
Quick fix is to replace `#include <EGL/eglextchromium.h>` with `#include <EGL/eglext_angle.h>` in xbmc/windowing/X11/GLContextEGL.cpp but we need to be backwards compatible.

However adding a cmake test fails to "find" eglext_angle.h as the cmake compile test fails due to EGLAPIENTRYP being undefined. The eglext_angle.h is missing an include for eglplatform.h (which xbmc/windowing/X11/GLContextEGL.h does so via including EGL/eglext.h which is probably normal setup). Cannot see a way to get cmake to create a compile test that includes both headers. Trying check_include_files just tests each header individually. (And check_include_file also gives an annoying cmake CMP0075 warning.)

```
diff -ur xbmc/cmake/modules/FindEGL.cmake xbmc-eglextheader/cmake/modules/FindEGL.cmake
--- xbmc/cmake/modules/FindEGL.cmake
+++ xbmc-eglextheader/cmake/modules/FindEGL.cmake
@@ -9,6 +9,7 @@
 # EGL_INCLUDE_DIRS - the EGL include directory
 # EGL_LIBRARIES - the EGL libraries
 # EGL_DEFINITIONS - the EGL definitions
+# HAVE_EGLEXTANGLE - if eglext_angle.h exists else use eglextchromium.h
 #
 # and the following imported targets::
 #
@@ -35,6 +36,11 @@
   set(EGL_LIBRARIES ${EGL_LIBRARY})
   set(EGL_INCLUDE_DIRS ${EGL_INCLUDE_DIR})
   set(EGL_DEFINITIONS -DHAS_EGL=1)
+  include(CheckIncludeFile)
+  check_include_file("EGL/eglext_angle.h" HAVE_EGLEXTANGLE)
+  if(HAVE_EGLEXTANGLE)
+    list(APPEND EGL_DEFINITIONS "-DHAVE_EGLEXTANGLE=1")
+  endif()

   if(NOT TARGET EGL::EGL)
     add_library(EGL::EGL UNKNOWN IMPORTED)
diff -ur xbmc/xbmc/windowing/X11/GLContextEGL.h xbmc-eglextheader/xbmc/windowing/X11/GLContextEGL.h
--- xbmc/xbmc/windowing/X11/GLContextEGL.h
+++ xbmc-eglextheader/xbmc/windowing/X11/GLContextEGL.h
@@ -13,7 +13,11 @@
 #include "threads/CriticalSection.h"

 #include <EGL/eglext.h>
+#ifdef HAVE_EGLEXTANGLE
+#include <EGL/eglext_angle.h>
+#else
 #include <EGL/eglextchromium.h>
+#endif
 #include <X11/Xutil.h>

 class CGLContextEGL : public CGLContext
```


### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1.  Install git mesa
2.  Compile Kodi


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
From CMakeError.log with above check_include_file we get;
```
Determining if the include file EGL/eglext_angle.h exists failed with the following output:
Change Dir: CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/ninja cmTC_593f3 && [1/2] Building C object CMakeFiles/cmTC_593f3.dir/CheckIncludeFile.c.o
FAILED: CMakeFiles/cmTC_593f3.dir/CheckIncludeFile.c.o
/usr/bin/x86_64-pc-linux-gnu-gcc -DNDEBUG -Wall -o CMakeFiles/cmTC_593f3.dir/CheckIncludeFile.c.o
-c CMakeFiles/CMakeTmp/CheckIncludeFile.c
In file included from CMakeFiles/CMakeTmp/CheckIncludeFile.c:1:
/usr/include/EGL/eglext_angle.h:162:20: error: unknown type name ‘EGLAPIENTRYP’
  162 | typedef EGLBoolean(EGLAPIENTRYP PFNEGLCREATESTREAMPRODUCERD3DTEXTUREANGLEPROC)(EGLDisplay dpy, EGLStreamKHR stream, const EGLAttrib *attrib_list);
      |
In file included from CMakeFiles/CMakeTmp/CheckIncludeFile.c:1:
/usr/include/EGL/eglext_angle.h:163:20: error: unknown type name ‘EGLAPIENTRYP’
....REPEAT FOR ALL LINES WITH EGLAPIENTRYP...
ninja: build stopped: subcommand failed.
```


### Your Environment
Used Operating system: 
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [X] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Gentoo
 - Kodi version:19 and 20 affected.



<!--- End of this issue -->

--- a/cmake/modules/FindEGL.cmake
+++ b/cmake/modules/FindEGL.cmake
@@ -9,6 +9,7 @@
 # EGL_INCLUDE_DIRS - the EGL include directory
 # EGL_LIBRARIES - the EGL libraries
 # EGL_DEFINITIONS - the EGL definitions
+# HAVE_EGLEXTANGLE - if eglext_angle.h exists else use eglextchromium.h
 #
 # and the following imported targets::
 #
@@ -35,6 +36,11 @@ if(EGL_FOUND)
   set(EGL_LIBRARIES ${EGL_LIBRARY})
   set(EGL_INCLUDE_DIRS ${EGL_INCLUDE_DIR})
   set(EGL_DEFINITIONS -DHAS_EGL=1)
+  include(CheckIncludeFiles)
+  check_include_files("EGL/egl.h;EGL/eglext.h;EGL/eglext_angle.h" HAVE_EGLEXTANGLE)
+  if(HAVE_EGLEXTANGLE)
+    list(APPEND EGL_DEFINITIONS "-DHAVE_EGLEXTANGLE=1")
+  endif()
 
   if(NOT TARGET EGL::EGL)
     add_library(EGL::EGL UNKNOWN IMPORTED)

--- a/tools/Linux/kodi-standalone.sh.in
+++ b/tools/Linux/kodi-standalone.sh.in
@@ -30,21 +30,21 @@ LOOP=1
 CRASHCOUNT=0
 LASTSUCCESSFULSTART=$(date +%s)
 
-while [ $(( $LOOP )) = "1" ]
+while [ $LOOP -eq 1 ]
 do
   $APP
   RET=$?
   NOW=$(date +%s)
-  if [ $(( ($RET >= 64 && $RET <=66) || $RET == 0 )) = "1" ]; then # clean exit
+  if [ $RET -ge 64 ] && [ $RET -le 66 ] || [ $RET -eq 0 ]; then # clean exit
     LOOP=0
   else # crash
     DIFF=$((NOW-LASTSUCCESSFULSTART))
-    if [ $(($DIFF > 60 )) = "1" ]; then # Not on startup, ignore
+    if [ $DIFF -gt 60 ]; then # Not on startup, ignore
       LASTSUCESSFULSTART=$NOW
       CRASHCOUNT=0
     else # at startup, look sharp
       CRASHCOUNT=$((CRASHCOUNT+1))
-      if [ $(($CRASHCOUNT >= 3)) = "1" ]; then # Too many, bail out
+      if [ $CRASHCOUNT -ge 3 ]; then # Too many, bail out
         LOOP=0
         echo "${APP} has exited in an unclean state 3 times in the last ${DIFF} seconds."
         echo "Something is probably wrong"

--- a/tools/Linux/kodi.sh.in
+++ b/tools/Linux/kodi.sh.in
@@ -180,10 +180,10 @@ do
   CHILD=$!
   wait "${CHILD}"
   RET=$?
-  if [ $(( $RET == 65 )) = "1" ]
+  if [ $RET -eq 65 ]
   then # User requested to restart app
     LOOP=1
-  elif [ $(( ($RET >= 131 && $RET <= 136) || $RET == 139 )) = "1" ]
+  elif [ $RET -ge 131 ] && [ $RET -le 136 ] || [ $RET -eq 139 ]
   then # Crashed with core dump
     print_crash_report
   fi

--- a/xbmc/windowing/X11/GLContextEGL.h
+++ b/xbmc/windowing/X11/GLContextEGL.h
@@ -13,7 +13,11 @@
 #include "threads/CriticalSection.h"
 
 #include <EGL/eglext.h>
+#ifdef HAVE_EGLEXTANGLE
+#include <EGL/eglext_angle.h>
+#else
 #include <EGL/eglextchromium.h>
+#endif
 #include <X11/Xutil.h>
 
 class CGLContextEGL : public CGLContext

